core/src/cluster_slots.rs

```diff
@@ -0,0 +1,267 @@
+use crate::{cluster_info::ClusterInfo, epoch_slots::EpochSlots, serve_repair::RepairType};
+
+use solana_ledger::{bank_forks::BankForks, staking_utils};
+use solana_sdk::{clock::Slot, pubkey::Pubkey};
+
+use std::{
+    collections::{HashMap, HashSet},
+    rc::Rc,
+    sync::RwLock,
+};
+
+#[derive(Default)]
+pub struct ClusterSlots {
+    cluster_slots: HashMap<Slot, HashMap<Rc<Pubkey>, u64>>,
+    keys: HashSet<Rc<Pubkey>>,
+    since: Option<u64>,
+    default_peers: HashMap<Rc<Pubkey>, u64>,
```

Comment:
This is really a mapping from node id to total stake right? Maybe we should name it something more clarifying 😛